### PR TITLE
Ruby 3.3 support: add symbolize_names kwarg to MatchData#named_captures

### PR DESCRIPTION
**Pull request by evaniainbrooks**
issue #8029 

I am unsure if my usage of keywords/checkArity/optional/checkArgumentCount is correct. I couldn't find another example exactly like this in the codebase. It seems I can still call `named_captures(:bogus)` without any complaints so perhaps someone can point me in the right direction.

### CODE REVIEW

File: core/src/main/java/org/jruby/RubyMatchData.java [hunk 1]
```diff
@@ -846,27 +848,47 @@ public RubyFixnum hash() {
...
+    @JRubyMethod(keywords = true, checkArity = false, optional = 1)
+    public RubyHash named_captures(ThreadContext context, IRubyObject[] args) {
         check();
+
         Ruby runtime = context.runtime;
+        int argc = Arity.checkArgumentCount(runtime, args.length, 0, 0, true);
         RubyHash hash = RubyHash.newHash(runtime);
         if (regexp == context.nil) return hash;
 
+        IRubyObject symbolizeNamesValue = runtime.getNil();
```
Comment (enebo) [inline core/src/main/java/org/jruby/RubyMatchData.java:860]: I think this is probably better as a boolean and lower perform the isTrue when processing the option.

**Review by enebo — CHANGES_REQUESTED** [review]
The argument processing stuff has always been a bit tedious in JRuby.  The code itself is not that complicated but it feels like we could be doing this better.  Keywords in particular really need a rethink for native methods.

Just change the two things and this is in.

**Comment by enebo** [comment]
@evaniainbrooks I do not see my second message about argumenterror if optional arg is not a hash.  Do you see that comment?  I do not seem to be able to see the actual review,

**Comment by evaniainbrooks** [comment]
> @evaniainbrooks I do not see my second message about argumenterror if optional arg is not a hash.  Do you see that comment?  I do not seem to be able to see the actual review,

I did not see it either, but no worries, I will add this now.

**Comment by enebo** [comment]
Thanks!   CI is pretty weird atm.  I am trying to update the parser now too so we can at least pass more.

**Pull request merged**